src/server/cards/adventures/Warrior.js

```diff
@@ -0,0 +1,40 @@
+import Card from 'cards/Card';
+
+export default class Warrior extends Card {
+  static cost = new Card.Cost({ coin: 4 });
+  static types = new Set(['Action', 'Attack', 'Traveller']);
+  async onPlay(player, event) {
+    await player.draw(2);
+    const numTravellersInPlay = player.playArea.filter(card => card.types.has('Traveller')).length;
+    await player.forEachOtherPlayer(async other => {
+      if (event.handledByPlayer.get(other)) {
+        return;
+      }
+      for (let i = 0; i < numTravellersInPlay; i++) {
+        const [card] = other.lookAtTopOfDeck(1);
+        if (card) {
+          await other.discard(card, other.deck);
+          if (card.cost.potion === 0 && card.cost.debt === 0 && (card.cost.coin === 3 || card.cost.coin === 4)) {
```

Comment:
Card is only trashed if it is still the top card of the discardPile. Discarding a tunnel will gain a Gold on top stopping the trash.

This is part of the larger lose track rule, not sure if we should add checks for it here or if it needs to be somewhere bigger.